internal/k8s/controller.go

```diff
@@ -1408,22 +1408,25 @@ func getStatusFromEventTitle(eventTitle string) string {
 		return conf_v1.StateWarning
 	case "AddedOrUpdated", "Updated":
 		return conf_v1.StateValid
-	default:
-		return "Unknown"
 	}
 }
 
-func (lbc *LoadBalancerController) updateVirtualServerVirtualServerRouteStatusesFromEvents() error {
-	vss := lbc.getVirtualServers()
+func (lbc *LoadBalancerController) updateVirtualServersStatusFromEvents() error {
 	var allErrs []error
-	for _, vs := range vss {
+	for _, obj := range lbc.virtualServerLister.List() {
```

Comment:
@lucacome the correct class will have to be handled  for this case in your PR as well https://github.com/nginxinc/kubernetes-ingress/pulls . Also for VirtualServerRoutes